src/test/scala/org/infinispan/spark/suites/StreamingSuite.scala

```diff
@@ -47,9 +47,8 @@ class StreamingSuite extends FunSuite with SparkStream with MultipleServers with
       stream.writeToInfinispan(getProperties)
 
       ssc.start()
-      ssc.awaitTerminationOrTimeout(2000)
 
-      cache.size shouldBe 3
+      waitForCondition(() => cache.size == 3)
```

Comment:
I think scalatest already has something built in for this:

http://www.artima.com/docs-scalatest-2.0.M5/org/scalatest/concurrent/Eventually.html